cluster-type/src/lib.rs

```diff
@@ -1,8 +1,10 @@
 #![cfg_attr(feature = "frozen-abi", feature(min_specialization))]
 #![cfg_attr(docsrs, feature(doc_auto_cfg))]
+#![cfg_attr(not(feature = "std"), no_std)]
```

Comment:
We've been declaring the crate no_std without the cfg check in other crates -- any reason to do this here? I'd prefer to keep it consistent
```suggestion
#![no_std]
```